### PR TITLE
INTMDB-254: Fix replication_specs behaviour when update cluster

### LINKED ISSUE #550: INVALID_CLUSTER_CONFIGURATION when adding new regions
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v1.0.5 on darwin_amd64
mongo provider: 1.0.0
```

### Terraform Output

```hcl
Terraform will perform the following actions:

  # mongodbatlas_cluster.runtime will be updated in-place
  ~ resource "mongodbatlas_cluster" "runtime" {
        id                                              = "********"
        name                                            = "prod"
        # (31 unchanged attributes hidden)



      - replication_specs {
          - id         = "******" -> null
          - num_shards = 1 -> null
          - zone_name  = "Zone 1" -> null

          - regions_config {
              - analytics_nodes = 0 -> null
              - electable_nodes = 3 -> null
              - priority        = 7 -> null
              - read_only_nodes = 0 -> null
              - region_name     = "EUROPE_WEST_4" -> null
            }
        }
      + replication_specs {
          + id         = (known after apply)
          + num_shards = 1
          + zone_name  = "Zone 1"

          + regions_config {
              + analytics_nodes = 0
              + electable_nodes = 0
              + priority        = 0
              + read_only_nodes = 2
              + region_name     = "SOUTHEASTERN_ASIA_PACIFIC"
            }
          + regions_config {
              + analytics_nodes = 0
              + electable_nodes = 3
              + priority        = 7
              + read_only_nodes = 0
              + region_name     = "EUROPE_WEST_4"
            }
        }
        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
mongodbatlas_cluster.runtime: Modifying... [id=****]
╷
│ Error: error updating MongoDB Cluster (runtime-prod): PATCH https://cloud.mongodb.com/api/atlas/v1.0/groups/*******/clusters/prod: 400 (request "INVALID_CLUSTER_CONFIGURATION") The specified cluster configuration is not valid.
│
│   with mongodbatlas_cluster.runtime,
│   on main.tf line 23, in resource "mongodbatlas_cluster" "runtime":
│   23: resource "mongodbatlas_cluster" "runtime" {
│
```

### Steps to Reproduce

1. Create a replicaset cluster
2. Add a new read only region replica

### Expected Behavior

New replica is added to the cluster or a meaningful error message appears

### PR DESCRIPTION
**Pull request by abner-dou**
## Description

A validation was re-coded in replication_specs cluster to get the original id when update the cluster

Link to any related issue(s):[INTMDB-254](https://jira.mongodb.org/browse/INTMDB-254)

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
TEST: 
-----------------------------------------------------
--- PASS: TestAccResourceMongoDBAtlasCluster_MultiRegion (1572.79s)
PASS
coverage: 8.1% of statements
ok      github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas 1575.224s       coverage: 8.1% of statements
?       github.com/mongodb/terraform-provider-mongodbatlas/version      [no test files]

### CODE REVIEW

**Review by themantissa — APPROVED** [review]
LGTM, thank you for knocking this out quickly (also tests haven't passed yet but will check again before merge - still two approvers to go)

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by thetonymaster — APPROVED** [review]
LGTM

**Pull request merged**
